### PR TITLE
Added ConvertFrom-DscResourceInstance fix #71

### LINKED ISSUE #71: Add a Convert-ObjectToHashtable to use as a comparison
The `Compare-DscParameterState` function accepts hashtable for comparing the CurrentValues with the DesiredValues.
The `DscResource.Common` module does not have the equivalent of `ConvertTo-Hashtable` for PowerShell classes (this command only accepts `[CimInstances[]]` as returned by MOF Based resources).

We should probably add the following function instead of adding it to every module with Class Based resources.
(like JeaDsc)
```
function Convert-ObjectToHashtable
{
    [OutputType([hashtable])]
    param (
        [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
        [object]$Object
    )

    process
    {
        $hashtable = @{ }

        foreach ($property in $Object.PSObject.Properties.Where({ $_.Value }))
        {
            $hashtable.Add($property.Name, $property.Value)
        }

        $hashtable
    }
}
```

### PR DESCRIPTION
**Pull request by NicolasBn**
#### Pull Request (PR) description
- Added cmdlet `ConvertFrom-DscResourceInstance` wich can be used to convert any object
 to in another type of format. It accepts objects from pipeline.

#### This Pull Request (PR) fixes the following issues
    - Fixes #71 
    - Fixes [Issue 246 on Sampler](https://github.com/gaelcolas/Sampler/issues/246)

#### Task list

- [x] Added an entry to the change log under the Unreleased section of the
      file CHANGELOG.md. Entry should say what was changed and how that
      affects users (if applicable), and reference the issue being resolved
      (if applicable).
- [ ] Documentation added/updated in README.md.
- [x] Comment-based help added/updated for all new/changed functions.
- [ ] Localization strings added/updated in all localization files as appropriate.
- [ ] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] Integration tests added/updated (where possible). See
  [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

---
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/dscresource.common/72)

### CODE REVIEW

File: source/Public/Convert-ObjectToHashtable.ps1 [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    [OutputType([Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true,ValueFromPipeline=$true)]
+        [PSObject]
+        $InputObject
+    )
+    begin {
+
+    }
```
Comment (gaelcolas) [inline source/Public/Convert-ObjectToHashtable.ps1:55]:
```suggestion
    )
```

File: source/Public/Convert-ObjectToHashtable.ps1 [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+
+#>
+function Convert-ObjectToHashtable
+{
+    [CmdletBinding()]
+    [OutputType([Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true,ValueFromPipeline=$true)]
```
Comment (gaelcolas) [inline source/Public/Convert-ObjectToHashtable.ps1:49]:
```suggestion
        [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
```

File: source/Public/Convert-ObjectToHashtable.ps1 [hunk 3]
```diff
@@ -0,0 +1,74 @@
+
+<#
+    .SYNOPSIS
+        Convert any object to hashtable.
+
+    .DESCRIPTION
+        This function is used to convert an psobject into a hashtable.
+
+    .PARAMETER InputObject
+        The object that should convert to hashtable.
```
Comment (gaelcolas) [inline source/Public/Convert-ObjectToHashtable.ps1:10]:
```suggestion
        The object that should be converted to hashtable.
```

File: source/Public/Convert-ObjectToHashtable.ps1 [hunk 4]
```diff
@@ -0,0 +1,74 @@
+
+<#
+    .SYNOPSIS
+        Convert any object to hashtable.
+
+    .DESCRIPTION
+        This function is used to convert an psobject into a hashtable.
```
Comment (gaelcolas) [inline source/Public/Convert-ObjectToHashtable.ps1:7]:
```suggestion
        This function is used to convert a PSObject into a hashtable.
```

File: source/Public/Convert-ObjectToHashtable.ps1 [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        FirstName = 'Peter'
+        LastName = 'Smith'
+    }
+
+    $ObjectArray | Convert-ObjectToHashtable
+
+    This creates pscustomobjects and converts there properties/values to Hashtable Keys/Values through the pipeline.
+
+
+
```
Comment (gaelcolas) [inline source/Public/Convert-ObjectToHashtable.ps1:41]:
```suggestion

```

File: source/Public/Convert-ObjectToHashtable.ps1 [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            }
+
+            return $hashResult
+        }
+    }
+
+    end {
+
+    }
+
```
Comment (gaelcolas) [inline source/Public/Convert-ObjectToHashtable.ps1:73]:
```suggestion
```

**Review by gaelcolas — CHANGES_REQUESTED** [review]
Thanks, a few update to make.

**Comment by NicolasBn** [comment]
Based on suggestion of @gaelcolas, I aggregate the issue #71  and [issue 246 of Sample](https://github.com/gaelcolas/Sampler/issues/246).
I rename the function to `ConvertFrom-DscResourceInstance`.

File: source/Public/ConvertFrom-DscResourceInstance.ps1 [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [PSObject]
+        $InputObject,
+
+        [Parameter()]
+        [ValidateSet('HashTable')]
+        [String]
+        $OutPutFormat = 'HashTable'
+
+    )
```
Comment (gaelcolas) [inline source/Public/ConvertFrom-DscResourceInstance.ps1:59]:
```suggestion
    )

```

**Review by gaelcolas — APPROVED** [review]

**Comment by johlju** [comment]
We should have updated documentation in the README.md as well. I add an issue to track it.

**Pull request merged**